umapi_client/functional.py

```diff
@@ -111,31 +82,13 @@ def __init__(self, id_type=IdentityTypes.adobeID, email=None, username=None, dom
             raise ArgumentError("Identity type (%s) must be one of %s" % (id_type, [i.name for i in IdentityTypes]))
         self.id_type = id_type
         self.email = None
-        self.domain = None
-        if username:
-            if email and username.lower() == email.lower():
-                # ignore the username if it's the same as the email (policy default)
-                username = None
-            elif id_type is not IdentityTypes.federatedID:
-                raise ArgumentError("Username must match email except for Federated ID")
-            else:
-                self._validate(username=username)
-                if domain:
-                    self._validate(domain=domain)
-                    self.domain = domain
-        if email:
-            self._validate(email=email)
+        self.domain = domain
```

Comment:
you need to also add `self.username = username` here.

Reply:
`self.username` is not set in the current version of the init function.

https://github.com/adobe-apiplatform/umapi-client.py/blob/master/umapi_client/functional.py#L87

I can add it.

Reply:
Just checked and self.username isn't expected or used anywhere in Action or UserAction.  Why do we need to set it?

Reply:
Interesting.  I guess just passing it to `Action.__init__` is enough, because that puts it in the user specification of the action.

I was expecting us to save it and then on "create" check that the username being specified was the same as in the init (just as we do with the email field).  But now I realize that `create` doesn't take a username at all :).